src/hooks/useUrlState.ts

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { filtersFromQuery } from '../attribute-filters';
+import { AttributeList, Filters } from '../components/filters/filter.types';
+import { useQueryParams } from './useQueryParams';
+
+interface UseUrlState {
+  defaultQuery?: string;
+  attributes: AttributeList;
+}
+
+const QUERY_PARAM_KEY = 'q';
+const TENANT_PARAM_KEY = 'tenant';
+const SHOW_RESOURCES_PARAM_KEY = 'showResources';
+const DEFAULT_TENANT = 'application';
+const DEFAULT_SHOW_RESOURCES = '0';
+const DEFAULT_QUERY = '{ log_type =~ ".+" } | json';
+
+export const useUrlState = ({
+  defaultQuery = DEFAULT_QUERY,
+  attributes,
+}: UseUrlState) => {
+  const queryParams = useQueryParams();
+  const history = useHistory();
+  const location = useLocation();
+
+  const initialQuery = queryParams.get(QUERY_PARAM_KEY) ?? defaultQuery;
+  const initialTenant = queryParams.get(TENANT_PARAM_KEY) ?? DEFAULT_TENANT;
+
+  const [query, setQuery] = React.useState(initialQuery);
+  const [tenant, setTenant] = React.useState(initialTenant);
+  const [filters, setFilters] = React.useState<Filters | undefined>({});
+  const [areResourcesShown, setAreResourcesShown] = React.useState(false);
+
+  const setTenantInURL = (tenant: string) => {
+    queryParams.set(TENANT_PARAM_KEY, tenant);
+    history.push(`${location.pathname}?${queryParams.toString()}`);
+  };
+
+  const setShowResourcesInURL = (showResources: boolean) => {
+    queryParams.set(SHOW_RESOURCES_PARAM_KEY, showResources ? '1' : '0');
+    history.push(`${location.pathname}?${queryParams.toString()}`);
+  };
+
+  const setQueryInURL = (newQuery: string) => {
+    queryParams.set(QUERY_PARAM_KEY, newQuery);
+    history.push(`${location.pathname}?${queryParams.toString()}`);
+  };
+
+  const setValuesFromParams = (urlParams: URLSearchParams) => {
```

Comment:
It's a bit confusing having `queryParams` and `urlParams`, which are basically the same. Is it possible to use `queryParams` here?